opengrok-indexer/src/main/java/org/opengrok/indexer/history/MercurialHistoryParser.java

```diff
@@ -185,8 +185,8 @@ public void processStream(InputStream input) throws IOException {
                      String[] move = part.split(" \\(");
                      File f = new File(mydir + move[0]);
                      if (!move[0].isEmpty() && f.exists() &&
-                         !renamedFiles.contains(move[0])) {
-                             renamedFiles.add(move[0]);
+                         !renamedFiles.contains(repository.getDirectoryNameRelative() + File.separator + move[0])) {
```

Comment:
What about changing the `renamedFiles` to `Set`? In that case we would not need the `contains` check and it should be also faster.